wit/couchbase/document.wit

```diff
@@ -0,0 +1,347 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Support CRUD access and modification of Documents stored in a Couchbase cluster.
+///
+/// Reference: https://docs.couchbase.com/c-sdk/current/concept-docs/documents.html
+interface document {
+  use types.{
+      document, document-id, document-error, mutation-metadata, time, durability-level, retry-strategy, request-span,
+      replica-read-level
+  };
+
+  /////////////////////////
+  /// Document - Insert ///
+  /////////////////////////
+
+  /// Options for performing a document insert
+  record document-insert-options {
+    /// When the document should expire (nanoseconds)
+    expires-in-ns: u64,
+
+    /// (Deprecated)
+    /// Traditional durability settings for persistence, only usable from Couchbase versions earlier than 6.0
+    ///
+    /// See: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#1.6@go-sdk::durability.adoc
+    persist-to: u64,
+
+    /// (Deprecated)
+    /// Traditional durability settings for replication, only usable from Couchbase versions earlier than 6.0
+    ///
+    /// See: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#1.6@go-sdk::durability.adoc
+    replicate-to: u64,
+
+    /// Durability level for the mutation
+    ///
+    /// (this should be used instead of persist-to/replicate-to)
+    durability-level: durability-level,
+
+    /// Timeout that should be used, in nanoseconds
+    /// Even if a timeout is not specified, the implementer *may* provide a default timeout.
+    timeout-ns: option<u64>,
+
+    /// How and whether to retry the operation
+    retry-strategy: option<retry-strategy>,
+
+    /// A known span to associate this lookup with
+    parent-span: option<request-span>,
+  }
+
+  /// Insert a document with a new ID
+  insert: func(
+    id: document-id,
+    doc: document,
+    options: option<document-insert-options>,
+  ) -> result<mutation-metadata, document-error>;
+
+  //////////////////////////
+  /// Document - Replace ///
+  //////////////////////////
+
+  /// Options for performing a document replace
+  record document-replace-options {
+    /// CAS revision of the document
+    revision: u64,
```

Comment:
I've just noticed the use of the word `revision` instead of `cas`. I'm concerned about the confusion it could bring with the `revid` (or even `seqno`) document metadata on Couchbase Server and the `revision id` on Sync Gateway / Couchbase Mobile.
IMHO it would be a safer choice to name this option `cas` everywhere to alleviate any possible ambiguity.

Reply:
Ah I see the chance for confusion there -- it's not too hard to make the replacements so I'll go ahead and do that!

- [ ] rename `revision:` to `cas:`